Library/include/CSP/Common/Map.h

```diff
@@ -114,7 +114,15 @@ template <typename TKey, typename TValue> class CSP_API Map
 			return *this;
 		}
 
-		Container->~MapType();
+		if (Container)
+		{
+			Container->~MapType();
+		}
+		else
+		{
+			Container = (MapType*) csp::memory::DllAlloc(sizeof(MapType));
```

Comment:
When could `Container == nullptr` at the time where the `=` operator is invoked? A cursory glace at the constructors suggest it should always have been allocated at this point?

Reply:
It was crashing for union assignment

Reply:
I assume because the constructor hasnt been called and it was assigning to empty memory

Reply:
Ah, I see.